staging/src/k8s.io/kubectl/pkg/cmd/clusterinfo/clusterinfo_dump.go

```diff
@@ -181,7 +172,33 @@ func (o *ClusterInfoDumpOptions) Run() error {
 		}
 	}
 
-	if err := o.PrintObj(nodes, setupOutputWriter(o.OutputDir, o.Out, "nodes", fileExtension)); err != nil {
+	// When writing json to stdout, we need to wrap the entire output in []
```

Comment:
Agreed that kubectl doesn't normally format output like this.  The reason for this change is that `kubectl cluster-info dump` makes many individual requests, which each return valid JSON responses, and kubectl concatenates all of the responses together (without being in an array).  The concatenations of all JSON responses isn't valid JSON.

At first, I thought that just removing the message that said "Cluster info dumped to standard output" would fix the problem and make it valid JSON, but it turns out this was also a cause of invalid JSON.

While I was working on this, I wondered if kubectl cluster-info dump should even try to emit valid JSON, or if it should just do a raw dump and not make any promises about it being valid JSON.  I think it is implied that it would be valid JSON, but I don't think anywhere it makes that guarantee.

I am open to whatever change makes sense to keep with the kubectl way of doing things